flow/FileTraceLogWriter.cpp

```diff
@@ -87,11 +88,28 @@ void FileTraceLogWriter::write(const std::string& str) {
 void FileTraceLogWriter::open() {
 	cleanupTraceFiles();
 
-	auto finalname = format("%s.%d.%s", basename.c_str(), ++index, extension.c_str());
+	++index;
+
+	// this allows one process to write 10 billion log files
+	// this should be enough - if not we could make the base larger...
+	ASSERT(index > 0 && index < 10);
+
+	int indexWidth = int(::floor(log10f(float(index)) + 1.0));
+	char indexWidthC;
+	UNSTOPPABLE_ASSERT(indexWidth < 62); // we don't use `0` so we can only use 61 chars
+	if (indexWidth < 10) {
+		indexWidthC = '0' + indexWidth;
+	} else if (indexWidth < 36) {
+		indexWidthC = 'A' + (indexWidth - 10);
+	} else {
+		indexWidthC = 'a' - (indexWidth - 36);
```

Comment:
Is this meant to be a plus sign?

Reply:
hehe - yep. As AJ pointed out, this is dead code, but I should still fix it...